sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/utils/AvroUtils.java

```diff
@@ -601,12 +601,26 @@ private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundE
     private Map<String, String> getMapping(Schema schema) {
       Map<String, String> mapping = Maps.newHashMap();
       for (Field field : schema.getFields()) {
-        String underscore = CaseFormat.LOWER_UNDERSCORE.to(CaseFormat.LOWER_CAMEL, field.getName());
-        mapping.put(underscore, field.getName());
+        String fieldName = field.getName();
+        String getter;
+        if (fieldName.contains("_")) {
+          if (Character.isLowerCase(fieldName.charAt(0))) {
+            // field_name -> fieldName
+            getter = CaseFormat.LOWER_UNDERSCORE.to(CaseFormat.LOWER_CAMEL, fieldName);
+          } else {
+            // FIELD_NAME -> fIELDNAME
+            getter = CaseFormat.UPPER_CAMEL.to(CaseFormat.LOWER_CAMEL, fieldName.replace("_", ""));
```

Comment:
```suggestion
            getter = CaseFormat.UPPER_UNDERSCORE.to(CaseFormat.LOWER_CAMEL, fieldName.replace("_", ""));
```

Reply:
When Avro compiles a schema to a Java class, it will take a field named `FIELD_NAME` and create a getter named `getFIELDNAME`.  So this method needs to produce getter field name `fIELDNAME`.

After some trial and error, I found I could produce this if I remove the underscore and convert it from `UPPER_CAMEL`.  When I use `UPPER_UNDERSCORE` (with or without the underscore in the field name), it produces a getter field name of `fieldName` or `fieldname`, which won't map to the `getFIELDNAME` method.

Reply:
Sure, I will add tests for all of the cases as soon as I have time.  Hopefully within one or two weeks.

Reply:
I have added an extended comment for this conversion and updated the existing test Avro schema to cover the new mappable field name styles.  Can you please confirm that this is acceptable?